From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 3 Sep 2010 21:40:35 -0400
Subject: [scsi] qla2xxx: add IS_QLA82XX check in update_fw_options
Message-id: <20100903214035.2767.17787.sendpatchset@localhost.localdomain>
Patchwork-id: 28149
O-Subject: [RHEL 5.6 PATCH 7/9] qla2xxx: Added check for IS_QLA82XX in
	update_fw_options.
Bugzilla: 619814
RH-Acked-by: Rob Evers <revers@redhat.com>

Bugzilla
--------

Bug 619814 (https://bugzilla.redhat.com/show_bug.cgi?id=619814)

Upstream Status
---------------

Will be submitted upstream with our next patch submission.

Description
-----------

>From d9c6d2fb3fd8196d98c6a2585e7fa8c954e5ae5e Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Thu, 29 Jul 2010 12:30:14 +0530
Subject: [PATCH] qla2xxx: Added check for IS_QLA82XX in update_fw_options.


diff --git a/drivers/scsi/qla2xxx/qla_init.c b/drivers/scsi/qla2xxx/qla_init.c
index 19b1d75..9a5b6ab 100644
--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -1259,6 +1259,9 @@ qla24xx_update_fw_options(scsi_qla_host_t *ha)
 {
 	int rval;
 
+	if (IS_QLA82XX(ha))
+		return;
+
 	/* Update Serial Link options. */
 	if ((le16_to_cpu(ha->fw_seriallink_options24[0]) & BIT_0) == 0)
 		return;
